### PR TITLE
Add param ShouldGenPassword for API "POST /api/admin/users"

### PR DESCRIPTION
**Pull request by supercharlesliu**
Issue Link: #14093 

Add param `is_thirdpart ` in `POST /api/admin/users` to indicate this user comes from a thirdpart LDAP/SSO.
A third-party user can be created without providing a password because users can log in with their LDAP.

### CODE REVIEW

**Comment by torkelo** [comment]
LDAP can already create users on the fly when they login.

File: pkg/api/dtos/user.go [hunk 1]
```diff
@@ -14,10 +14,11 @@ type SignUpStep2Form struct {
...
 type AdminCreateUserForm struct {
-	Email    string `json:"email"`
-	Login    string `json:"login"`
-	Name     string `json:"name"`
-	Password string `json:"password" binding:"Required"`
+	Email         string `json:"email"`
+	Login         string `json:"login"`
+	Name          string `json:"name"`
+	Password      string `json:"password"`
+	ThirdPartUser bool   `json:"is_thirdpart"`
```
Comment (torkelo) [inline pkg/api/dtos/user.go]: Would not ShouldGenPassword be more descriptive. Oh, and underscores in json is no go

Comment (supercharlesliu) [reply pkg/api/dtos/user.go]: Yeah, `ShouldGenPassword` is better.

**Comment by supercharlesliu** [comment]
> LDAP can already create users on the fly when they login.

We still need to create users to sign roles and permissions, It's happened the initialization process before users can login.

**Comment by torkelo** [comment]
Think generating a random password can easily be done in your init script. So this logic does not need to be in Grafana.

**Pull request closed without merging**